backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -18,39 +18,234 @@
+            # Check each service
+            for service in enabled_services_dict.keys():
+                # If the service has permissions that need to be denied
+                if service in service_disable_permissions:
+                    # If the service is deactivated
+                    if not enabled_services_dict[service]:
+                        if "ResourceType" in service_disable_permissions[service]:
+                            suspend_all_of_type(service_disable_permissions[service]["ResourceType"])
+                        if "Statements" in service_disable_permissions[service]:
+                            deny_policy_statements.extend(service_disable_permissions[service]["Statements"])
+
+            # Check for permissions associated with multiple services
+            for group in service_group_disable_permissions:
+                # If all the services in a group are deactivated, then add the group deny statements to the deny policy
+                if "ServiceGroup" in group:
+                    active_service_in_group = False
+                    for group_service in group["ServiceGroup"]:
+                        if enabled_services_dict[group_service]:
+                            active_service_in_group = True
+                            break
+                    # Deny permissions related to the service group
+                    if not active_service_in_group and "Statements" in group:
+                        deny_policy_statements.extend(group["Statements"])
+
+            # When all services are activated, apply a filler statement so that the policy does not need to be unattached and deleted
+            # Unattaching from all roles could cause throttling and will require re-attaching to all dynamic roles later
+            if len(deny_policy_statements) == 0:
+                deny_policy_statements.extend(filler_deny_statement)
+
+            # Will create, update the current deny services policy with the new permissions
+            iam_manager.update_dynamic_policy(
+                iam_manager.generate_policy_string(deny_policy_statements),
+                "app-denied-services",
+                "services",
+                "deny",
+                on_create_attach_to_notebook_role=True,
+            )
+
+        except Exception as e:
+            logging.exception(f"Failed to update the denied services policy {str(e)}")
+            response_status_code = 207
+            warning_issues.append("Failed when updating the deny policy for deacivating service IAM permissions.")
+
+        # All updates were successfully executed
+        if response_status_code == 200:
+            response_message = f"Successfully updated app configuration"
+        # App config was successfullyl updated but other elements may have failed resulting in mixed results
+        elif response_status_code == 207:
+            response_message = "Successfully updated app configuration, but some issues were encountered."
```

Comment:
An update should be atomic - either everything succeeds, or everything fails. 

The response should be something like:

> An error was encountered while activating/deactivating services. Please contact your system administrator for assistance.

Reply:
After further discussion with @dustins and @douglas1850 we are going to wait to merge @dustins [PR](https://github.com/awslabs/mlspace/pull/118) and then will move forward with creating a mostly atomic action for handling this.

Reply:
I also just merged my change with the SystemRole so I would advise for leveraging that on the app config update lambda. That way you can stop/terminate/delete resources *after* we've done all of our checks.